### PR TITLE
Implement new mod callback api

### PR DESCRIPTION
**Pull request by raonygamer**

### CODE REVIEW

File: resources/modinfo.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "nameId": "Example",
     "name": "Template Name",
     "description": "Template Description",
-    "version": "1.0.0",
+    "version": "1.0.1",
```
Comment (ThePixelGamer) [inline resources/modinfo.json]: The version bump isn't needed

File: src/Example.cpp [hunk 2]
```diff
@@ -2,14 +2,24 @@
...
-
-MOD_FUNCTION Zenova::Mod* CreateMod() {
-	return new Example;
+// Called when the Mod is loaded
+MOD_FUNCTION void ModLoad(ModContext& ctx) {
+	Zenova_Info("Mod {} loaded!", ctx.folder);
+}
+
+// Called after Minecraft main
+// todo: implement above (currently called after ~2 seconds)
```
Comment (ThePixelGamer) [inline src/Example.cpp]: can you remove the todo: comments

**Review by ThePixelGamer — COMMENTED** [review]
LGTM, just some minor nitpicks

File: src/Example.cpp (hunk 2 above)
Comment (raonygamer) [reply src/Example.cpp]: oke

File: resources/modinfo.json (hunk 1 above)
Comment (raonygamer) [reply resources/modinfo.json]: oke

**Comment by raonygamer** [comment]
ready?

**Pull request merged**